File: ActivityLifeCycle/app/src/main/java/com/example/activitylifecycle/MainActivity.java
package com.example.activitylifecycle;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {


    Button button;
    Button button2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        button=findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                goToSecondActivity();
            }
        });

        button2=findViewById(R.id.button2);
        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openWebPage();
            }
        });

        Toast.makeText(this, "onCreate() is called ", Toast.LENGTH_SHORT).show();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.my_menu,menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        int itemid=item.getItemId();

        if(itemid==R.id.action_bar_home){
            Toast.makeText(this, "You Selected Home", Toast.LENGTH_SHORT).show();
        }else if(itemid==R.id.action_search){
            Toast.makeText(this, "You selected Search", Toast.LENGTH_SHORT).show();
        }
        return super.onOptionsItemSelected(item);
    }

    //Explicit Intent
    public  void goToSecondActivity(){
        Intent intent=new Intent(this, MainActivity2.class);
        startActivity(intent);
    }

    //Implicit Intent
    public void openWebPage(){
        Uri webpage= Uri.parse("https://github.com/yugbit2021");

        Intent intent=new Intent(Intent.ACTION_VIEW,webpage);
        startActivity(intent);
    }















    @Override
    protected void onStart() {
        super.onStart();
        Toast.makeText(this, "onStart() is called ", Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onResume() {
        super.onResume();
        Toast.makeText(this, "onResume() is called ", Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onPause() {
        super.onPause();
        Toast.makeText(this, "onPause() is called ", Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onStop() {
        super.onStop();
        //Release resources , save data , etc
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Toast.makeText(this, "onRestart() is called ", Toast.LENGTH_SHORT).show();
    }
}

File: AcitivityLifeCycle/app/src/main/java/com/example/acitivitylifecycle/MainActivity.kt
package com.example.acitivitylifecycle

import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btn1:Button=findViewById(R.id.button);
        btn1.setOnClickListener(){
            //Explicit Intent
            var i: Intent = Intent(this,Activity2::class.java);

            //Passing data b/w activities
            //Key , value
            //key should be string
            i.putExtra("age",30);
            startActivity(i);
        }

        val btn2:Button=findViewById(R.id.button2);
        btn2.setOnClickListener(){
            //Implicit Intent
            var intent:Intent=Intent(Intent.ACTION_VIEW);
            intent.setData(Uri.parse("https://www.google.com"));
            startActivity(intent);
        }

    }
}